src/expr/src/scalar/func.rs

```diff
@@ -4210,6 +4214,33 @@ fn trim_whitespace<'a>(a: Datum<'a>) -> Datum<'a> {
     Datum::from(a.unwrap_str().trim_matches(' '))
 }
 
+fn right<'a>(a: Datum<'a>, b: Datum<'a>) -> Datum<'a> {
+    let string: &'a str = a.unwrap_str();
+    let n = b.unwrap_int64();
+
+    // the index of every char in bytes
+    let byte_indices = string.char_indices().map(|(i, _)| i);
+
+    let start_in_bytes = if n > 0 {
+        // pop off the last n-1 elements
+        let mut byte_indices = byte_indices.rev().skip((n - 1) as usize);
```

Comment:
It's generally better to use Rust's [`TryFrom` Trait](https://doc.rust-lang.org/std/convert/trait.TryFrom.html) instead of `as` to perform these conversions (and bubble up any errors). We don't do this everywhere ([but we want to!](https://materializeinc.slack.com/archives/CM7ATT65S/p1614821217253300)). This would look like:
```
usize::try_from(n-1)
```
There are other `try_from` calls in this file for reference!

Reply:
Another thing to do, if you have the spirit, is to track down what postgres does in this corner case. We'd error with `try_from`, but it could be that postgres does something specific like correctly handle strings of ridonkulous length.

Reply:
Thanks for being so thorough with me! 🍵 I'm still learning most of this Rust business.